app/modals/revoke-delegation/revoke-delegation-modal.tsx

```diff
@@ -1,238 +1,66 @@
-        <DecryptWalletForm
-          description="Enter your password to revoke delegation"
-          onSetPassword={password => setPassword(password)}
-          onForgottenPassword={() => {
-            closeModal();
-            history.push(routes.SETTINGS);
-          }}
-          hasSubmitted={hasSubmitted}
-          decryptionError={decryptionError}
-        />
-      ),
-      footer: (
-        <StackingModalFooter>
-          <StackingModalButton mode="tertiary" onClick={closeModal}>
-            Close
-          </StackingModalButton>
-          <StackingModalButton
-            isLoading={isDecrypting || isSendingTx}
-            isDisabled={isDecrypting || isSendingTx}
-            onClick={() => broadcastTx()}
-          >
-            Revoke Delegation
-          </StackingModalButton>
-        </StackingModalFooter>
-      ),
-    }),
-    [RevokeDelegationModalStep.SignWithLedgerAndSend]: () => ({
-      header: (
-        <StackingModalHeader onSelectClose={closeModal}>Confirm on your Ledger</StackingModalHeader>
-      ),
-      body: <SignTxWithLedger step={ledgerStep} isLocked={isLocked} ledgerError={null} />,
-      footer: (
-        <StackingModalFooter>
-          <StackingModalButton
-            mode="tertiary"
-            onClick={() => {
-              setHasSubmitted(false);
-              closeModal();
-            }}
-          >
-            Close
-          </StackingModalButton>
-          <StackingModalButton
-            isDisabled={
-              hasSubmitted || ledgerStep !== LedgerConnectStep.ConnectedAppOpen || isLocked
-            }
-            isLoading={hasSubmitted}
-            onClick={() => void broadcastTx()}
-          >
-            Sign transaction
-          </StackingModalButton>
-        </StackingModalFooter>
-      ),
-    }),
-
-    [RevokeDelegationModalStep.FailedContractCall]: () => ({
-      header: <StackingModalHeader onSelectClose={closeModal} />,
-      body: (
-        <StackingFailed walletType={walletType}>Failed to call stacking contract</StackingFailed>
-      ),
-      footer: (
-        <StackingModalFooter>
-          <StackingModalButton mode="tertiary" onClick={closeModal}>
-            Close
-          </StackingModalButton>
-          <StackingModalButton onClick={() => setStep(initialStep)}>Try again</StackingModalButton>
-        </StackingModalFooter>
-      ),
-    }),
-  };
-
-  const { header, body, footer } = txFormStepMap[step]();
+  const revokeDelegation = useCallback(
+    (signedTx: StacksTransaction) =>
+      broadcastTx({
+        onSuccess: async txId => {
+          await safeAwait(watchForNewTxToAppear({ txId, nodeUrl: api.baseUrl }));
+          await refetchMempool();
+          dispatch(homeActions.closeRevokeDelegationModal());
+        },
+        onFail: (error?: any) => {
+          if (error) setNodeResponseError(error);
+        },
+        tx: signedTx,
+      }),
+    [api.baseUrl, broadcastTx, dispatch, refetchMempool]
```

Comment:
should `watchForNewTxToAppear`, `homeActions` be included in deps? seems like there are a few missing, but might be wrong if the eslint rule hasn't yelled

Reply:
Both of these are es imports, their reference will never change, so afaik don't need to be listed as deps.